src/video/SDL_surface.c

```diff
@@ -142,6 +143,10 @@ SDL_CreateSurface(int width, int height, Uint32 format)
             return NULL;
         }
 
+        if (SDL_ISPIXELFORMAT_FOURCC(surface->format->format)) {
+            size = SDL_SW_SizeYUV(surface->format->format, surface->w, surface->h, &surface->pitch);
```

Comment:
Should check for a zero size here.

Reply:
current code doesn't check that for RGB surface. so not sure?